server/src/main/java/org/opensearch/action/admin/cluster/remotestore/repository/RemoteStoreRepositoryRegistrationHelper.java

```diff
@@ -0,0 +1,146 @@
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.remotestore.repository;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.RepositoriesMetadata;
+import org.opensearch.cluster.metadata.RepositoryMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.repositories.RepositoriesService;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * RemoteStore Repository Registration helper
+ */
+public class RemoteStoreRepositoryRegistrationHelper {
+
+    public static boolean isRemoteStoreEnabled(ClusterState currentState) {
+        return currentState.getMetadata().settings().get("cluster.remote_store.segment.repository") != null
+            && currentState.getMetadata().settings().get("cluster.remote_store.translog.repository") != null;
+    }
+
+    private static Settings buildSettings(String stringSettings) {
+        Settings.Builder settings = Settings.builder();
+
+        String[] stringKeyValue = stringSettings.split(",");
+        for (int i = 0; i < stringKeyValue.length; i++) {
+            String[] keyValue = stringKeyValue[i].split(":");
+            settings.put(keyValue[0].trim(), keyValue[1].trim());
+        }
+
+        return settings.build();
+    }
+
+    public static RepositoryMetadata buildSegmentRepositoryMetadata(DiscoveryNode node) {
+
+        String name = node.getAttributes().get("remote_store.segment.repository.name");
+        String type = node.getAttributes().get("remote_store.segment.repository.type");
+        String stringSettings = node.getAttributes().get("remote_store.segment.repository.settings");
+
+        assert name != null : "Name cannot be null";
+        assert type != null : "Type cannot be null";
+        assert stringSettings != null : "Settings cannot be null";
+
+        return new RepositoryMetadata(name, type, buildSettings(stringSettings));
+
+    }
+
+    public static RepositoryMetadata buildTranslogRepositoryMetadata(DiscoveryNode node) {
+        String name = node.getAttributes().get("remote_store.translog.repository.name");
+        String type = node.getAttributes().get("remote_store.translog.repository.type");
+        String stringSettings = node.getAttributes().get("remote_store.translog.repository.settings");
+
+        assert name != null : "Name cannot be null";
+        assert type != null : "Type cannot be null";
+        assert stringSettings != null : "Settings cannot be null";
+
+        return new RepositoryMetadata(name, type, buildSettings(stringSettings));
+    }
+
+    synchronized public static ClusterState validateOrAddRemoteStoreRepository(ClusterState currentState, DiscoveryNode node) {
```

Comment:
`synchronized` isn't needed as the execute of the ClusterStateTaskExecutor is guaranteed to be executed in isolation